message/infrastructure/googlecloud/subscriber.go

```diff
@@ -218,6 +218,10 @@ func (s *Subscriber) receive(
 			return
 		}
 
+		msgCtx, cancel := context.WithCancel(ctx)
```

Comment:
it's interesting why tests were passing, did you investigated it?

Reply:
IIRC, they weren't. Nobody checked since.
Tests in CI recommended :)